source/button.js

```diff
@@ -18,69 +17,88 @@ class LikelyButton {
     constructor(widget, likely, options) {
         this.widget = widget;
         this.likely = likely;
-        this.options = merge(options);
-        this.serviceName = this.detectService();
+        this.options = mergeToNew(options);
+        this.detectService();
+        if (this.isRecognized()) {
+            this.detectParams();
```

Comment:
Allows to gracefully ignore unrecognized buttons, in case they're added manually.